packages/zoe/src/typeGuards.js

```diff
@@ -79,3 +121,231 @@ export const InvitationElementShape = M.splitRecord({
+    getTerms: M.call().returns(M.split(TermsShape, M.record())),
+    getIssuers: M.call().returns(IssuerKeywordRecordShape),
+    getBrands: M.call().returns(BrandKeywordRecordShape),
+    getInstallationForInstance: M.call().returns(InstallationShape),
+    getInvitationIssuer: M.call().returns(IssuerShape),
+
+    saveIssuer: M.call(IssuerShape, M.string()).returns(M.promise()),
+    makeZoeMint: M.call(KeywordShape)
+      .optional(AssetKindShape, DisplayInfoShape, M.pattern())
+      .returns(M.or(ZoeMintI, M.remotable('zoeMint'), M.promise())),
+    registerFeeMint: M.call(KeywordShape, M.remotable('feeMintAccess')).returns(
+      M.remotable('feeMint'),
+    ),
+    getInstanceRecord: M.call().returns(InstanceRecordShape),
+    getIssuerRecords: M.call().returns(M.arrayOf(IssuerRecordShape)),
+    getWithdrawFacet: M.call().returns(M.remotable('WithdrawFacet')),
+    initInstanceAdmin: M.call(
+      InstanceHandleShape,
+      M.remotable('instanceAdmin'),
+    ).returns(M.promise()),
+    deleteInstanceAdmin: M.call(InstanceAdminI).returns(),
+    makeInvitation: M.call(InvitationHandleShape, M.string())
+      .optional(M.any(), M.pattern())
+      .returns(M.promise()),
+    getRoot: M.call().returns(M.any()),
+    getAdminNode: M.call().returns(M.remotable('adminNode')),
+  }),
+  withdrawFacet: M.interface('WithdrawFacet', {
+    withdrawPayments: M.call(AmountKeywordRecordShape).returns(
+      PaymentPKeywordRecordShape,
+    ),
+  }),
+});
+
+// In `SwingSet/src/types-external.js`, it's typed as `*`.
+const BundleCapShape = M.any();
+
+export const ZoeStorageManagerIKit = harden({
+  zoeServiceDataAccess: M.interface('ZoeService dataAccess', {
+    getTerms: M.call(InstanceHandleShape).returns(
+      M.split(TermsShape, M.record()),
+    ),
+    getIssuers: M.call(InstanceHandleShape).returns(IssuerKeywordRecordShape),
+    getBrands: M.call(InstanceHandleShape).returns(BrandKeywordRecordShape),
+    getInstallationForInstance: M.call(InstanceHandleShape).returns(
+      M.or(M.promise(), M.remotable('Installation')),
+    ),
+    getInvitationIssuer: M.call().returns(IssuerShape),
+
+    getBundleIDFromInstallation: M.call(M.any()).returns(M.promise()),
+    installBundle: M.call(M.any()).returns(M.promise()),
+    installBundleID: M.call(M.string()).returns(M.promise()),
+
+    getPublicFacet: M.call(InstanceHandleShape).returns(
+      M.or(M.promise(), M.remotable('PublicFacet')),
+    ),
+    getOfferFilter: M.call(InstanceHandleShape).returns(M.arrayOf(M.string())),
+    setOfferFilter: M.call(
+      InstanceHandleShape,
+      M.arrayOf(M.string()),
+    ).returns(),
+    getProposalShapeForInvitation: M.call(InvitationHandleShape).returns(
+      M.or(M.pattern(), M.undefined()),
+    ),
+  }),
+  makeOfferAccess: M.interface('ZoeStorage makeOffer access', {
+    getAssetKindByBrand: M.call(BrandShape).returns(AssetKindShape),
+    installBundle: M.call(InstanceHandleShape).returns(),
```

Comment:
Is this more correct?

```suggestion
    installBundle: M.call(InstanceHandleShape).returns(M.promise()),
```

Reply:
yep

Reply:
even better. Resolving.